### ISSUE TITLE
uint64_t should be std::

### ISSUE THREAD

**Issue by VictorEijkhout** [mfem/mfem#4423]
gcc14 complains:

```
/work/00434/eijkhout/mfem/mfem-4.4/general/hash.hpp:1001:39: error: ISO C++ forbids declaration of ‘
  3) pmix/3.1.4     6) TACC            9) impi/19.0.9    12) phdf5/1.10.4  15) boost/1.72           │type name’ with no type [-fpermissive]
export CC=mpicc                                                                                     │ 1001 |       auto k = reinterpret_cast<const uint64_t &>(*begin);
export CXX=mpicxx                                                                                   │      |                                       ^~~~~~~~
export FC=mpif90                                                                                    │/work/00434/eijkhout/mfem/mfem-4.4/general/hash.hpp:1001:39: error: expected ‘>’ before ‘uint64_t’
#HYPRE                                                                                              │/work/00434/eijkhout/mfem/mfem-4.4/general/hash.hpp:1001:39: error: expected ‘(’ before ‘uint64_t’
==========                                                                                          │ 1001 |       auto k = reinterpret_cast<const uint64_t &>(*begin);
mkdir hypre_built                                                                                   │      |                                       ^~~~~~~~
git clone git@github.com:hypre-space/hypre.git                                                      │      |                                       (
```

I think that should be `std::uint64_t`.

**Comment by tzanio** [comment]
Thanks for reporting this.

We have `uint64_t` in multiple places in the code, so I wonder if we should replace it everywhere or typedef `uint64_t` if it is not defined.

**Comment by VictorEijkhout** [comment]
This problems seems to be fixed in 4.7; I was using 4.4.

**Comment by v-dobrev** [comment]
Yes, this issue was fixed in #3630 and it was first present in a release in `v4.6`.
